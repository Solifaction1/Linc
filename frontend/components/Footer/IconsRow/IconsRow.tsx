/* eslint-disable react/require-default-props */
import React from 'react'
import { useTranslation } from 'next-i18next'
import { useMediaQuery } from 'react-responsive'
import SwitchIcon from '@templates/SwitchIcon'
import { Icon } from '@templates/switch-icon.config'
import { Navigation } from '@models/navigation.model'
import SwitchCTA from '@templates/SwitchCTA'
import { BREAKPOINTS } from '@constants/breakpoints'
import styles from './icons-row.module.scss'

const IconsRow = (props: {
	contactItem: Navigation | undefined
	facebookLink: string
	linkedInLink: string
	className?: string
}) => {
	const { t } = useTranslation('common')
	const isTablet = useMediaQuery({ maxWidth: BREAKPOINTS.lg })

	return (
		<div className={`${styles.iconsRow} ${props.className}`}>
			{props.contactItem && (
				<div className={styles.link}>
					<SwitchCTA
						ctaData={{
							title: props.contactItem.title,
							link: props.contactItem.path ?? '',
							externalLink: false,
							buttonType: 'Button primary',
						}}
					/>
				</div>
			)}

			{!isTablet && <p className={styles.text}>{t('follow_us')}</p>}

			{props.linkedInLink && (
				<a href={props.linkedInLink}>
					<SwitchIcon icon={Icon.LINKEDIN} className={styles.icon} />
				</a>
			)}

			{props.facebookLink && (
				<a href={props.facebookLink}>
					<SwitchIcon icon={Icon.FACEBOOK} className={styles.icon} />
				</a>
			)}
		</div>
	)
}

export default IconsRow
