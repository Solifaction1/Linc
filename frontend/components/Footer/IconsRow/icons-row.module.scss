@import "styles/vars";
@import "styles/utilities";

.iconsRow {
    display: flex;
    justify-content: flex-end;
    max-height: 5.2rem;

    @media (max-width: $break-md) {
        justify-content: flex-start;
        margin-bottom: calc(#{$spacing-base} * 8);
    }

    .link {
        margin-right: calc(#{$spacing-base} * 4);
        @media (max-width: $break-md) {
            margin-right: calc(#{$spacing-base} * 2);
        }

        button {
            border: 1px solid $color-beige;
        }

        p {
            color: $color-beige !important;
            font-size: $font-size-text-md;
        }
    }

    .text {
        margin-right: calc(#{$spacing-base} * 2);

        color: $color-beige;

        white-space: pre-line;
        line-height: 2.4rem;
    }

    .icon {
        width: 4.2rem;
        height: 4.2rem;
        margin-right: calc(#{$spacing-base} * 2);
        padding: 1rem;
        border: 1px solid $color-beige;
        border-radius: 50%;

        fill: $color-black;

        path {
            fill: $color-beige;
        }

        &:hover {
            background: $color-beige-dark-opacity-25;
        }
    }
}
