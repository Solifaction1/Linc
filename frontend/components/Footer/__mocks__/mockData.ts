/* eslint-disable max-lines */
import { Contact } from "@models/contact.model"
import { Navigation, RelatedContactInformation } from "@models/navigation.model"

const mockNavigationParent1: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 1,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'parent1',
    related: null,
    sitemap_exclude: true,
    title: 'Parent 1',
    type: '',
    uiRouterKey: 'parent1',
    updatedAt: ''
}

const mockNavigationParent2: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 4,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'parent2',
    related: null,
    sitemap_exclude: true,
    title: 'Parent 2',
    type: '',
    uiRouterKey: 'parent2',
    updatedAt: ''
}


export const mockContact: Contact = {
    id: 10,
    street: 'street',
    houseNumber: 'houseNumber',
    postalCode: 0,
    city: 'city',
    telephone: '+1234567890'
}

const mockContactInformationRelated: RelatedContactInformation = {
    createdAt: 'createdAt',
    facebook: 'facebook',
    id: 7,
    instagram: 'instagram',
    linkedin: 'linkedin',
    locale: 'en',
    navigationItemId: 0,
    publishedAt: 'publishedAt',
    sitemap_exclude: false,
    updatedAt: 'updatedAt',
    updatedBy: 'updatedBy',
    __contentType: '__contentType'
}

const mockContactInformation: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 8,
    items: null,
    menuAttached: true,
    order: 0,
    parent: null,
    path: 'contact-information',
    related: mockContactInformationRelated,
    sitemap_exclude: true,
    title: 'Contact Information',
    type: '',
    uiRouterKey: 'contact-information',
    updatedAt: ''
}

const mockNavigationDataCompliantLinks: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 2,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'data-compliant-links',
    related: null,
    sitemap_exclude: true,
    title: 'Data Compliant Links',
    type: '',
    uiRouterKey: 'data-compliant-links',
    updatedAt: ''
}

export const mockNavigationData = [
    mockNavigationParent1,
    mockNavigationParent2,
    mockContactInformation,
    mockNavigationDataCompliantLinks,
]