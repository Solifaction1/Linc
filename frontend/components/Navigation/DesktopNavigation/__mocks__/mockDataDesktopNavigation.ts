/* eslint-disable max-lines */
import { Navigation } from "@models/navigation.model";

export const navigationMockData: Navigation[] = [{
    audience: [],
    collapsed: false,
    createdAt: 'createdAt',
    id: 1,
    menuAttached: false,
    order: 0,
    path: 'path',
    sitemap_exclude: false,
    title: 'title',
    type: 'type',
    uiRouterKey: 'uiRouterKey',
    updatedAt: 'updatedAt',
    items: 'items',
    parent: null,
    related: null
}]
