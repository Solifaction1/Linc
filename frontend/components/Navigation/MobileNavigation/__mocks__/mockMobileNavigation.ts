/* eslint-disable max-lines */
import { Navigation } from "@models/navigation.model";

export const navigationMockData: Navigation[] = [{
    audience: [],
    collapsed: false,
    createdAt: 'createdAt',
    id: 1,
    menuAttached: false,
    order: 0,
    path: 'path',
    sitemap_exclude: false,
    title: 'title',
    type: 'type',
    uiRouterKey: 'uiRouterKey',
    updatedAt: 'updatedAt',
    children: [],
    items: 'items',
    parent: null,
    related: {
        createdAt: '',
        facebook: 'https://www.facebook.com',
        linkedin: 'https://www.linkedin.com',
        instagram: '',
        locale: '',
        id: 1,
        navigationItemId: 2,
        publishedAt: '',
        sitemap_exclude: false,
        updatedAt: '',
        updatedBy: '',
        __contentType: ''
    }
}]
