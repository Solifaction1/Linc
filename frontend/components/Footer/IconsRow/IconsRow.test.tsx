import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { navigationMockData } from '@components/Navigation/DesktopNavigation/__mocks__/mockDataDesktopNavigation'
import IconsRow from './IconsRow'

jest.mock('next/router', () => ({
	useRouter() {
		return {
			locale: 'locale',
		}
	},
}))

jest.mock('next-i18next', () => ({
	useTranslation() {
		return {
			t: () => {
				'test'
			},
		}
	},
}))

describe('IconsRow', () => {
	it('renders no icons when no props are given IconsRow', () => {
		const { container } = render(
			<IconsRow
				contactItem={navigationMockData[0]}
				facebookLink=""
				linkedInLink="linkedInLink"
			/>
		)

		expect(container).toMatchSnapshot()

		expect(container.getElementsByClassName('iconsRow')[0]).toBeVisible()

		expect(container.querySelector('a[href="facebookLink"]')).toBeNull()

		expect(container.querySelector('a[href="linkedInLink"]')).toBeVisible()
		expect(container.querySelectorAll('svg')[0]).toBeVisible()
	})

	it('renders icons when props are given', () => {
		const { container } = render(
			<IconsRow
				contactItem={navigationMockData[0]}
				facebookLink="facebookLink"
				linkedInLink="linkedInLink"
			/>
		)

		expect(container).toMatchSnapshot()

		expect(container.getElementsByClassName('iconsRow')[0]).toBeVisible()

		expect(container.querySelector('a[href="facebookLink"]')).toBeVisible()
		expect(container.querySelectorAll('svg')[0]).toBeVisible()

		expect(container.querySelector('a[href="linkedInLink"]')).toBeVisible()
		expect(container.querySelectorAll('svg')[1]).toBeVisible()
	})
})
