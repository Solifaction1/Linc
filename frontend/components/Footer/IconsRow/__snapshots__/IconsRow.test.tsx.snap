// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconsRow renders icons when props are given 1`] = `
<div>
  <div
    class="iconsRow undefined"
  >
    <div
      class="link"
    >
      <button
        class="button false primary md false  "
        type="button"
      >
        <p
          class="label"
        >
          title
        </p>
      </button>
    </div>
    <p
      class="text"
    />
    <a
      href="linkedInLink"
    >
      <svg
        class="icon"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.47,2H3.53A1.45,1.45,0,0,0,2.06,3.43V20.57A1.45,1.45,0,0,0,3.53,22H20.47a1.45,1.45,0,0,0,1.47-1.43V3.43A1.45,1.45,0,0,0,20.47,2ZM8.09,18.74h-3v-9h3ZM6.59,8.48h0a1.56,1.56,0,1,1,0-3.12,1.57,1.57,0,1,1,0,3.12ZM18.91,18.74h-3V13.91c0-1.21-.43-2-1.52-2A1.65,1.65,0,0,0,12.85,13a2,2,0,0,0-.1.73v5h-3s0-8.18,0-9h3V11A3,3,0,0,1,15.46,9.5c2,0,3.45,1.29,3.45,4.06Z"
        />
      </svg>
    </a>
    <a
      href="facebookLink"
    >
      <svg
        class="icon"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.12,5.32H17V2.14A26.11,26.11,0,0,0,14.26,2C11.54,2,9.68,3.66,9.68,6.7V9.32H6.61v3.56H9.68V22h3.68V12.88h3.06l.46-3.56H13.36V7.05C13.36,6,13.64,5.32,15.12,5.32Z"
        />
      </svg>
    </a>
  </div>
</div>
`;

exports[`IconsRow renders no icons when no props are given IconsRow 1`] = `
<div>
  <div
    class="iconsRow undefined"
  >
    <div
      class="link"
    >
      <button
        class="button false primary md false  "
        type="button"
      >
        <p
          class="label"
        >
          title
        </p>
      </button>
    </div>
    <p
      class="text"
    />
    <a
      href="linkedInLink"
    >
      <svg
        class="icon"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.47,2H3.53A1.45,1.45,0,0,0,2.06,3.43V20.57A1.45,1.45,0,0,0,3.53,22H20.47a1.45,1.45,0,0,0,1.47-1.43V3.43A1.45,1.45,0,0,0,20.47,2ZM8.09,18.74h-3v-9h3ZM6.59,8.48h0a1.56,1.56,0,1,1,0-3.12,1.57,1.57,0,1,1,0,3.12ZM18.91,18.74h-3V13.91c0-1.21-.43-2-1.52-2A1.65,1.65,0,0,0,12.85,13a2,2,0,0,0-.1.73v5h-3s0-8.18,0-9h3V11A3,3,0,0,1,15.46,9.5c2,0,3.45,1.29,3.45,4.06Z"
        />
      </svg>
    </a>
  </div>
</div>
`;
